includes/Core/Util/Debug_Data.php

```diff
@@ -0,0 +1,272 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Util\Debug_Data
+ *
+ * @package   Google\Site_Kit\Core\Util
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Util;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\Authentication\Clients\OAuth_Client;
+use Google\Site_Kit\Core\Authentication\Credentials;
+use Google\Site_Kit\Core\Modules\Module;
+use Google\Site_Kit\Core\Modules\Module_With_Debug_Fields;
+use Google\Site_Kit\Core\Modules\Modules;
+use Google\Site_Kit\Core\Storage\Encrypted_Options;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+
+/**
+ * Class for integrating debug information with Site Health.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Debug_Data {
+	/**
+	 * Context instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Context
+	 */
+	private $context;
+
+	/**
+	 * Modules instance.
+	 *
+	 * @since n.e.x.t
+	 * @var Modules
+	 */
+	private $modules;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context $context Context instance.
+	 */
+	public function __construct( Context $context ) {
+		$this->context = $context;
+		$this->modules = new Modules( $context );
```

Comment:
Let's pass the basic class instances here like we do in other class constructors, rather than instantiating manually here and in individual methods below. They should be all optional, but mostly for testing. The `Plugin` class should always provide all of the instances. The following class instances should be (optionally) passed here, in addition to `Context`:
* `Options`
* `User_Options`
* `Authentication`
* `Modules`